sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -325,200 +329,202 @@ export class BatchingReceiverLite {
-
-          // Setting drain must be accompanied by a flow call (aliased to addCredit in this case).
-          receiver.drain = true;
-          receiver.addCredit(1);
-        } else {
-          cleanupBeforeResolveOrReject();
-
-          logger.verbose(
-            `${loggingPrefix} Resolving receiveMessages() with ${brokeredMessages.length} messages.`
-          );
-          resolve(brokeredMessages);
-        }
-      };
-
-      // Action to be performed on the "message" event.
-      const onReceiveMessage: OnAmqpEventAsPromise = async (context: EventContext) => {
-        // TODO: this appears to be aggravating a bug that we need to look into more deeply.
-        // The same timeout+drain sequence should work fine for receiveAndDelete but it appears
-        // to cause problems.
-        if (this._receiveMode === "peekLock") {
-          if (brokeredMessages.length === 0) {
-            // We'll now remove the old timer (which was the overall `maxWaitTimeMs` timer)
-            // and replace it with another timer that is a (probably) much shorter interval.
-            //
-            // This allows the user to get access to received messages earlier and also gives us
-            // a chance to have fewer messages internally that could get lost if the user's
-            // app crashes in receiveAndDelete mode.
-            if (totalWaitTimer) clearTimeout(totalWaitTimer);
-            const remainingWaitTimeInMs = getRemainingWaitTimeInMs();
-            totalWaitTimer = setTimeout(() => {
-              logger.verbose(
-                `${loggingPrefix} Batching, waited for ${remainingWaitTimeInMs} milliseconds after receiving the first message.`
-              );
-              finalAction();
-            }, remainingWaitTimeInMs);
-          }
-        }
+    const onError: OnAmqpEvent = (context: EventContext) => {
+      const eventType = context.session?.error != null ? "session_error" : "receiver_error";
+      let error = context.session?.error || context.receiver?.error;
 
-        try {
-          const data: ServiceBusMessageImpl = this._createServiceBusMessage(context);
-          if (brokeredMessages.length < args.maxMessageCount) {
-            brokeredMessages.push(data);
-          }
-        } catch (err) {
-          const errObj = err instanceof Error ? err : new Error(JSON.stringify(err));
-          logger.logError(
-            err,
-            `${loggingPrefix} Received an error while converting AmqpMessage to ServiceBusMessage`
-          );
-          reject(errObj);
-        }
-        if (brokeredMessages.length === args.maxMessageCount) {
-          finalAction();
-        }
-      };
+      if (error) {
+        error = translateServiceBusError(error);
+        logger.logError(error, `${loggingPrefix} '${eventType}' event occurred. Received an error`);
+      } else {
+        error = new ServiceBusError("An error occurred while receiving messages.", "GeneralError");
+      }
+      reject(error);
```

Comment:
Why don't we resolve for receive_and_delete here?

Reply:
I'm also curious but don't think you need to change this as part of this PR since this is existing behavior separate from your change.

Reply:
Yeah, just wanted to open up a discussion :)

Reply:
Issue created here to discuss further: https://github.com/Azure/azure-sdk-for-js/issues/13093